test/common/router/router_test.cc

```diff
@@ -6253,6 +6253,28 @@ TEST_F(RouterTest, ConnectExplicitTcpUpstream) {
   router_.onDestroy();
 }
 
+TEST_F(RouterTest, PostExplicitTcpUpstream) {
```

Comment:
Maybe also test that if the config is not present, that newStream is called on the TCP pool?

Reply:
is this what ConnectExplicitTcpUpstream does?

Reply:
Sorry, I misspoke.  I meant that with connect config and without allow_post == true, that POST results in the http pool.